zerver/views/user_settings.py

```diff
@@ -259,10 +259,14 @@ def json_change_settings(
             check_change_full_name(user_profile, full_name, user_profile)
 
     # Loop over user_profile.property_types
-    request_settings = {k: v for k, v in list(locals().items()) if k in user_profile.property_types}
+    request_settings = {
+        k: v
+        for k, v in list(locals().items())
+        if k in user_profile.property_types and k not in user_profile.notification_setting_types
```

Comment:
This is a bit odd and confusing as we included notification settings into `property_types` and then excluding them here because we have separate functions for each of them.

The major difference/barrier blocking us from being able to unify the functions is `do_change_notification_settings` creates a `RealmAuditLog` entry but the other doesn't. And then we are also sending legacy events `update_global_notifications` and `update_display_settings` from those functions but that should be easy to handle with a few `if`, `else`s.

Any ideas here? Is it useful to create `RealmAuditLog` entries for other setting changes too?

Reply:
I discussed this with Tim a few days back, he said that we should have a RealmAuditLog entry for all the settings. But I think we can wait for Tim's confirmation on this as we would need to decide whether we want to have a new event type which will be used for both settings or we will use the type used for notification setting for all the settings.